### PR TITLE
[travis] New travis job for CHECK_DOCS steps

### LINKED ISSUE #11004: New Travis job for CHECK_DOCS steps
This is a request to change how the Travis job is configured for Bitcoin. I'd submit a PR to do this myself, but the change can only be made by someone who has Travis configuration permissions for the bitcoin repo.

## Background

Currently there are a number of static-analysis style checks that are run in Travis, guarded by the `CHECK_DOC` environment variable. This variable is only set for the first test runner (the arm-linux-gnueabihf host), so the checks are only run once. All of these checks run very fast, and do not require building Bitcoin or running the regular C++/Python tests.

## Request

Configure a new Travis job that runs against PRs, and only runs the `CHECK_DOC` tests. At the same time, the `CHECK_DOC` stuff in the existing job would be removed.

## Motivation
Creating a separate Travis job for these tests would be an improvement because:
 * It would be more immediately obvious that a `CHECK_DOC` failure is related to one of the check scripts; currently it's confusing to new contributors (e.g. me) to see an ARM failure, when the failure is not actually related to any new architecture-specific code.
 * The job would either pass quickly or fail quickly, giving faster feedback to developers who submit PRs.
 * The main `.travis.yml` file in the repo could be cleaned up, by removing the `CHECK_DOC` guards in the `before_script` and `script` sections.

### PR DESCRIPTION
**Pull request by glaksmono**
Configure a new Travis job that runs against PRs, and only runs the `CHECK_DOC` tests. At the same time, the `CHECK_DOC` stuff in the existing job would be removed.

Fixes #11004

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -84,3 +70,24 @@ script:
 after_script:
     - echo $TRAVIS_COMMIT_RANGE
     - echo $TRAVIS_COMMIT_LOG
+
+jobs:
+  include:
+    - stage: CHECK_DOC
+      env:
+        - HOST=arm-linux-gnueabihf PACKAGES="g++-arm-linux-gnueabihf python3-pip shellcheck" DEP_OPTS="NO_QT=1" CHECK_DOC=1 GOAL="install" BITCOIN_CONFIG="--enable-glibc-back-compat --enable-reduce-exports"
```
Comment (MarcoFalke) [inline .travis.yml]: Please keep the actual arm cross build intact. The only thing you'd need to move here is the pip and shellcheck package

Also, would be nice if the check doc run before the tests, not the other way round.

Comment (glaksmono) [reply .travis.yml]: `HOST=arm-linux-gnueabihf` will be executed twice in the `check_doc` and `test` then?

**Comment by MarcoFalke** [comment]
You wouldn't need to set environment variables that are not used for the
check doc. I.e. no need for the cross compile HOST ...

File: .travis.yml [hunk 2]
```diff
@@ -84,3 +75,27 @@ script:
 after_script:
     - echo $TRAVIS_COMMIT_RANGE
     - echo $TRAVIS_COMMIT_LOG
+
+jobs:
+  include:
+    - stage: check_doc
+      sudo: required
+      env:
+        - PACKAGES="python3-pip shellcheck" CHECK_DOC=1
```
Comment (MarcoFalke) [inline .travis.yml]: Can remove CHECK_DOC here

File: .travis.yml [hunk 3]
```diff
@@ -84,3 +75,27 @@ script:
 after_script:
     - echo $TRAVIS_COMMIT_RANGE
     - echo $TRAVIS_COMMIT_LOG
+
+jobs:
+  include:
+    - stage: check_doc
+      sudo: required
```
Comment (MarcoFalke) [inline .travis.yml]: You can get rid of sudo here to speed up the boot-process:

```diff
diff --git a/.travis.yml b/.travis.yml
index f3f5305e4a..f547b3dc42 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -79,12 +79,13 @@ after_script:
 jobs:
   include:
     - stage: check_doc
-      sudo: required
-      env:
-        - PACKAGES="python3-pip shellcheck"
+      sudo: false
+      addons:
+       apt:
+        packages:
+         - python3-pip
+         - shellcheck
       install:
-        - if [ -n "$PACKAGES" ]; then travis_retry sudo apt-get update; fi
-        - if [ -n "$PACKAGES" ]; then travis_retry sudo apt-get install --no-install-recommends --no-upgrade -qq $PACKAGES; fi
         - if [ "$TRAVIS_EVENT_TYPE" = "pull_request" ]; then travis_retry pip3 install flake8 --user; fi
       before_script:
         - git fetch --unshallow

File: .travis.yml [hunk 4]
```diff
@@ -8,6 +8,9 @@ cache:
   - depends/built
   - depends/sdk-sources
   - $HOME/.ccache
+stages:
+  - check_doc
+  - test
 env:
   global:
     - MAKEJOBS=-j3
```
Comment (MarcoFalke) [inline .travis.yml]: Can remove the CHECK_DOC on the line below this one.

Comment (glaksmono) [reply .travis.yml]: Done

File: .travis.yml [hunk 5]
```diff
@@ -84,3 +74,26 @@ script:
 after_script:
     - echo $TRAVIS_COMMIT_RANGE
     - echo $TRAVIS_COMMIT_LOG
+
+jobs:
+  include:
+    - stage: check_doc
```
Comment (ken2812221) [inline .travis.yml]: How about add `sudo: false` here, travis can boot faster

**Comment by ken2812221** [comment]
Concept ACK

File: .travis.yml (hunk 5 above)
Comment (MarcoFalke) [reply .travis.yml]: Please see my previous diff to get it to work

Comment (MarcoFalke) [reply .travis.yml]: https://github.com/bitcoin/bitcoin/pull/13183#discussion_r186475302

**Comment by MarcoFalke** [comment]
Please squash your commits according to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

File: .travis.yml (hunk 5 above)
Comment (glaksmono) [reply .travis.yml]: Done

**Review by ken2812221 — APPROVED** [review]
utACK 372e8e95aaae449df3f7d2197e1b2f54288a1fbe

**Comment by MarcoFalke** [comment]
utACK 18c0b8430820114d14a15fe19b7ae86e62be17d8. Thanks for sticking with this. Indeed a very nice first contribution!

**Comment by glaksmono** [comment]
Thank you for all of the guidance @MarcoFalke

**Comment by sipa** [comment]
utACK 18c0b8430820114d14a15fe19b7ae86e62be17d8

**Comment by ken2812221** [comment]
re-utACK 18c0b8430820114d14a15fe19b7ae86e62be17d8

**Comment by practicalswift** [comment]
Thanks @glaksmono! Very nice contribution!

**Pull request merged**